client/my-sites/checkout/checkout-system-decider.js

```diff
@@ -124,34 +125,51 @@ export default function CheckoutSystemDecider( {
 	}
 
 	if ( 'composite-checkout' === checkoutVariant ) {
+		isLoggedOutCart &&
+			initGoogleRecaptcha(
+				'g-recaptcha',
+				'calypso/signup/pageLoad',
+				config( 'google_recaptcha_site_key' )
+			).then( ( result ) => {
+				if ( ! result ) {
+					return;
+				}
+
+				const { dispatch } = defaultRegistry;
+				dispatch( 'wpcom' ).setRecaptchaClientId( parseInt( result.clientId ) );
```

Comment:
I think I would do a `const { setRecaptchaClientId } = useDispatch( 'wpcom' );` at the root of the function and then you'd just be able to call `setRecaptchaClientId( ... )` inside this callback function.

Reply:
Done

Reply:
Interesting! I wonder if it's because there might be multiple versions of the core data packages installed, which might cause `useDispatch()` to use a different default registry ... hopefully not, I haven't experienced this issue before.

Reply:
I _think_ this is timing-related. When the component is first rendered, I think the store has not been created yet, so its actions don't exist. I've encountered this before in a few places, usually with selectors (which is why you'll see all the selectors in composite checkout use optional chaining). It's tricky. I'm ok with this solution.